contracts/consensus/Consensus.sol

```diff
@@ -512,17 +522,21 @@ contract Consensus is MasterCopyNonUpgradable, CoreStatusEnum, ConsensusI {
         // Validate the join params.
         validateJoinParams(_chainId, _core, _withdrawalAddress);
 
-        // Specified core must have creation status.
+        // Specified core must have genesis lifetime status.
         require(
-            isCore(_core),
-            "Core must be in an active state."
+            coreLifetime[_core] == CoreLifetime.creation,
+            "Core lifetime status must be creation."
         );
 
         // Stake in reputation contract.
         reputation.stake(msg.sender, _withdrawalAddress);
 
         // Join in core contract.
-        CoreI(_core).joinDuringCreation(msg.sender);
+        uint256 coreValidatorCount = CoreI(_core).joinDuringCreation(msg.sender);
```

Comment:
Proposal/Suggestion:

We can add `Core::minimumValidatorCountReached()` function.

Reply:
we will update this in a later task